storage/range.go

```diff
@@ -278,20 +279,76 @@ func (r *Range) IsFirstRange() bool {
 	return bytes.Equal(r.Desc().StartKey, engine.KeyMin)
 }
 
-// IsLeader returns true if this range replica is the raft leader.
-// TODO(spencer): this is always true for now.
-func (r *Range) IsLeader() bool {
-	return true
+// getLease returns the current leader lease.
+func (r *Range) getLease() *proto.Lease {
+	return (*proto.Lease)(atomic.LoadPointer(&r.lease))
+}
+
+// HasLeaderLease returns true if this range replica currently holds
+// the Raft leader lease. It's possible that the replica is not in
+// fact the leader, but holding the lease means that no other replica
```

Comment:
We might need to check HasLeaderLease at the beginning and end of any operation that depends on the lease being held, or else we may lose the lease while working (We could make HasLeaderLease return false if the lease is close to expiring, but then we would have problems if anything that depended on the lease took longer than whatever margin we used there). 

I'm not sure yet whether we need it though; it may be the case that an instantaneous check for leadership is enough. I think that's true for reads, but I'm not sure for writes. In general, if a node loses its lease then MultiRaft will drop its proposals, but if it quickly regains the lease under a new term then we might have an operation begin under one term and commit under another, even though the leader state was wiped in between. Passing the term around would let us ensure that proposals cannot cross term boundaries like that.